### PR TITLE
Upgrade dependencies, ROS2 common interfaces

### PR DESCRIPTION
**Pull request by ds58**

### CODE REVIEW

File: ihmc-pub-sub-generator/build.gradle.kts [hunk 1]
```diff
@@ -2,7 +2,6 @@ plugins {
    id("us.ihmc.ihmc-build")
    id("us.ihmc.ihmc-ci") version "8.3"
    id("us.ihmc.ihmc-cd") version "1.26"
-   id("com.github.hierynomus.license") version "0.14.0"
```
Comment (ds58) [inline ihmc-pub-sub-generator/build.gradle.kts:5]: I removed this, it's really unnecessary

File: ros2-common-interfaces/build.gradle.kts [hunk 2]
```diff
@@ -15,14 +15,14 @@ val rclInterfacesPath = "src/main/vendor/rcl_interfaces"
 val rclInterfacesUrl = "https://github.com/ros2/rcl_interfaces.git"
 val commonInterfacesPath = "src/main/vendor/common_interfaces"
 val commonInterfacesUrl = "https://github.com/ros2/common_interfaces.git"
-val ros2Release = "0.8.0"
+val ros2Release = "2.2.0"
```
Comment (ds58) [inline ros2-common-interfaces/build.gradle.kts:18]: Latest ros2 release

File: ros2-library/build.gradle.kts [hunk 3]
```diff
@@ -23,5 +23,5 @@ testDependencies {
    api("us.ihmc:ihmc-commons-test:0.34.0")
    api("us.ihmc:ros2-common-interfaces:source")
    api("us.ihmc:ros2-msg-to-pubsub-generator-test:source")
-   api("com.google.guava:guava:18.0")
+   api("com.google.guava:guava:33.3.1-jre")
```
Comment (calvertdw) [inline ros2-library/build.gradle.kts]: necessary?

**Review by calvertdw — APPROVED** [review]

File: ros2-library/build.gradle.kts (hunk 3 above)
Comment (ds58) [reply ros2-library/build.gradle.kts]: is the concern we will have multiple versions of guava on the classpath? I can do an audit and make sure they all line up across repos.

Comment (ds58) [reply ros2-library/build.gradle.kts]: reverted

**Pull request merged**